src/main/java/com/google/sps/servlets/EmailServlet.java

```diff
@@ -0,0 +1,168 @@
+import java.io.UnsupportedEncodingException;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.Properties;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet that sends emails to the author and followers of an answered question or a commented
+ * answer.
+ */
+@WebServlet("/email")
+public class EmailServlet extends HttpServlet {
+
+  /**
+   * Sends a welcome email to the user after they register
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get type from query string.
+    String typeOfElement = request.getParameter("type");
+    // Get ID from query string and convert to int.
+    int elementId = Integer.parseInt(request.getParameter("elementId"));
+
+    // Create content for mail. Call functions to see which users have to be notified
+    // and get their emails concatenated in a string.
+    String userEmails = getUserEmails(getUsersToNotify(typeOfElement, elementId));
+    String subject = "Activity on Mintern!";
+    String message = "Dear mintern,\n" +
+                     "You have new notifications in Mintern!\n" + 
+                     "Feel free to login and check it at: " +
+                     "internship-platform-step-2020.appspot.com/ .\n\n" +
+                     "Best wishes!\n" + 
+                     "The Mintern Team";
+
+    // Declares objects necesssary for the mail.
+    Properties props = new Properties();
+    Session session = Session.getDefaultInstance(props, null);
+    try {
+      // Set content for mail.
+      Message msg = new MimeMessage(session);
+      msg.setFrom(new InternetAddress("mintern@internship-platform-step-2020.appspotmail.com",
+          "Mintern"));
+      // Parse the concatenated string to convert it to an array of addresses.
+      msg.addRecipients(Message.RecipientType.TO, InternetAddress.parse(userEmails));
+      msg.setSubject(subject);
+      msg.setText(message);
+      // Send email.
+      Transport.send(msg);
+    } catch (AddressException e) {
+      System.out.println("Failed to set email address.");
+    } catch (MessagingException e) {
+      System.out.println("Failed to send email.");
+    } catch (UnsupportedEncodingException e) {
+      System.out.println("Failed to encode email.");
+    }
+  }
+
+  /**
+   * Queries IDs of the author of the modified question/answer and its followers.
+   */
+  private List<Integer> getUsersToNotify(String type, int elementId) {
```

Comment:
Take a look at the Utility class - there should be some examples there that you can follow, so I think it would make sense to add this there as well!

Reply:
Done!